code/game/machinery/embedded_controller/airlock_controllers.dm

```diff
@@ -91,49 +36,34 @@
 		tag_chamber_sensor = given_tag_chamber_sensor
 	..()
 
-/obj/machinery/embedded_controller/radio/airlock/airlock_controller/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/embedded_controller/radio/airlock/airlock_controller/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "simple_airlock_console.tmpl", name, 470, 290)
+		ui = new(user, src, ui_key, "Airlock", name, 470, 290, master_ui, state)
```

Comment:
I would suggest a more descriptive name for this, and I also recommend `AirlockController.js` as the file name too.
```suggestion
		ui = new(user, src, ui_key, "AirlockController", name, 470, 290, master_ui, state)
```

Reply:
Siding with steel. Please change this, possibly to `ExternalAirlockController` since it managed atmos stuffs.